tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -760,13 +760,15 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
             if (!kind.isClass) {
               reportError(s"Cannot select $item of non-class $cls")
             } else {
-              maybeClass.foreach {
-                _.lookupField(item).fold[Unit] {
-                  reportError(s"Class $cls does not have a field $item")
-                } { fieldDef =>
-                  if (fieldDef.tpe != tree.tpe)
-                    reportError(s"Select $cls.$item of type "+
-                      s"${fieldDef.tpe} typed as ${tree.tpe}")
+              maybeClass.foreach { c =>
+                if (c.hasInstances) { // see #3060
```

Comment:
> Consider moving the reason why we do this inline here. IMO a better position than somewhere in an issue history.

Done.

> Do we need to support a "full check mode"?

It might be nice, but it will be hard to define what "full" really means. In general, our IR checker can only check IR that *links* to begin with. So in any case, it is necessary for DCE in general to kick in before the IR checker. Then the problem is that we won't be able to check *everything* in all cases. So where do we draw the line about what should be checked in "full" check mode?